### PR TITLE
Fixed move automation and iframe step iterating (closes #414, closes #418)

### LINKED ISSUE #414: Runner skips InIframe step if the page inside an iframe is reloaded before a target element appears
The step number is not decremented if the page inside an iframe is reloaded while waiting for an action target to appear. TestContextStorage's property `` 'actionTargetWaiting'`` isn't used when the test step runs inside an iframe.

### LINKED ISSUE #418: Browsers hang during the test execution, when trying to perform an action after redirecting in an iframe
The reason of this behavior is infinite move automation that occurs when action performing.

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @AlexanderMoskovkin, @AndreyBelym

### CODE REVIEW

**Comment by AndreyBelym** [comment]
LGTM, also works for ASP tests.

**Comment by AlexanderMoskovkin** [comment]
lgtm

/cc @VasilyStrelyaev

File: test/functional/fixtures/regression/gh-414/test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+var expect = require('chai').expect;
+
+
+describe("Should restart a test step if redirect occurs while waiting for the action's target element", function () {
+    it('Should restart user-action after redirect', function () {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/gh-414/test.js]:
```
    it('Should restart a user action after redirect', function () {
```

File: test/functional/fixtures/regression/gh-414/testcafe-fixtures/index.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    '3.Click div "The third div"': function () {
+        act.click(document.getElementById('div3'));
+    },
+
+    '4.Assert': function () {
+        eq($('#result').text(), 'div3 click');
+    }
+};
+
+'@test'['Redirect before target element is appeared - should fail'] = {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/gh-414/testcafe-fixtures/index.test.js]:
```
'@test'['Redirect before the target element appears - should fail'] = {
```

File: test/functional/fixtures/regression/gh-414/test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+var expect = require('chai').expect;
+
+
+describe("Should restart a test step if redirect occurs while waiting for the action's target element", function () {
+    it('Should restart user-action after redirect', function () {
+        return runTests('testcafe-fixtures/index.test.js', 'Click on the element after redirect - should pass');
+    });
+
+    it("Should fail if action target element doesn't exist after redirect", function () {
+        return runTests('testcafe-fixtures/index.test.js', 'Redirect before target element is appeared - should fail', { shouldFail: true })
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/gh-414/test.js]: note that I renamed the test

File: test/functional/fixtures/regression/gh-418/testcafe-fixtures/index.test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+'@fixture GH-418 iframe page';
+'@page ./cross-domain.html';
+
+
+'@test'['Click in iframe on the element after redirect in iframe - should pass'] = {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/gh-418/testcafe-fixtures/index.test.js]:
```
'@test'['Click on the element in iframe after redirect occurs in the iframe - should pass'] = {
```

Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/gh-418/testcafe-fixtures/index.test.js]: can't find where this one is used

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by georgiy-abbasov** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Pull request merged**